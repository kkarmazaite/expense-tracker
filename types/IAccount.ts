export interface IAccount {
    id?: string
    userId: string
    name: string
  }