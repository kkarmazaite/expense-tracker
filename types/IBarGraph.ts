export interface IBarGraph {
    group: string 
    value: number 
    valueDisplay: string
  }