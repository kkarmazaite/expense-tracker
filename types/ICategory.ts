export type ICategoryAccountTypes = "income" | "expense"
export interface ICategory {
    id?: string
    accountId: string
    iconId?: string | null
    colorId?: string | null
    type: ICategoryAccountTypes
    name: string
  }