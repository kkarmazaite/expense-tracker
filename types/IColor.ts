export interface IColor {
    id?: string
    name: string
    hexCode: string
  }