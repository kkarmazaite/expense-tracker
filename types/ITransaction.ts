export interface ITransaction {
    id?: string
    categoryId: string
    date: Date
    amount: number
    description: string
  }