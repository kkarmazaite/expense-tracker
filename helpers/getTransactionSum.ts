import { ITransaction } from "~~/types/ITransaction"

export const getTransactionSum = (transactions:ITransaction[]) => {
  return transactions.reduce((sum, transaction:ITransaction) => sum + transaction.amount, 0)
}