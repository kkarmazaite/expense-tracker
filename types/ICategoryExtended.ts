import { ICategory } from "./ICategory";
import { ITransaction } from "./ITransaction";

export interface ICategoryExtented extends ICategory {
    transactions: ITransaction[]
   }