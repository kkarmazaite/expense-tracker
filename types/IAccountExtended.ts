import { IAccount } from "./IAccount";
import { ICategoryExtented } from "./ICategoryExtended";

export interface IAccountExtented extends IAccount {
    categories: ICategoryExtented[]
   }