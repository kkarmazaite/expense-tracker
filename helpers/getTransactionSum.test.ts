import { expect, describe, it } from 'vitest'
import { ITransaction } from '~~/types/ITransaction'
import { getTransactionSum } from './getTransactionSum'

describe('Helper | getTransactionSum', async() => {
  it('Check getTransactionSum function', () => {
    const input:ITransaction[] = [
      {
        id:"64393641fb31c3d4ab87beaa",
        date: new Date("2023-04-14T00:00:00.000Z"),
        amount:100,
        description:"",
        categoryId:"63da749ec2b86311ffab27a5",
        category: {
          accountId:"63da7427c2b86311ffab279f",
          color:null,
          colorId:null,
          icon:{
            id:"6426d6eaf4d0051801e1b45b",
            name:"tax",
          },
          iconId:"6426d6eaf4d0051801e1b45b",
          id:"63da749ec2b86311ffab27a5",
          name:"tax",
          type:"expense",

        },
      },
      {
        id:"64393641fb31c3d4ab87beaa",
        date: new Date("2023-04-14T00:00:00.000Z"),
        amount:200,
        description:"",
        categoryId:"63da749ec2b86311ffab27a5",
        category: {
          accountId:"63da7427c2b86311ffab279f",
          color:null,
          colorId:null,
          icon:{
            id:"6426d6eaf4d0051801e1b45b",
            name:"tax",
          },
          iconId:"6426d6eaf4d0051801e1b45b",
          id:"63da749ec2b86311ffab27a5",
          name:"tax",
          type:"expense",

        },
      },
    ]
    const output:number = 300

    expect(getTransactionSum(input)).toEqual(output);
    expect(getTransactionSum(input)).toMatchSnapshot()
  })
})