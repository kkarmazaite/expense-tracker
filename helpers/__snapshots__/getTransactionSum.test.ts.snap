// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Helper | getTransactionSum > Check getTransactionSum function 1`] = `300`;
